From 32b5bb021b9c4f40b7571f355431b704f55cce36 Mon Sep 17 00:00:00 2001
From: Matthew Waters <matthew@centricular.com>
Date: Mon, 8 May 2023 13:40:01 +1000
Subject: [PATCH 22/77] gl: move gl(base)mixer to library

For use by applications/libraries outside of the opengl plugin.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4608>
---
 ext/gl/gstglmosaic.c                      | 2 ++
 ext/gl/gstglmosaic.h                      | 2 +-
 ext/gl/gstglstereomix.h                   | 4 ++--
 ext/gl/gstglvideomixer.h                  | 8 ++++----
 ext/gl/meson.build                        | 2 --
 {ext => gst-libs/gst}/gl/gstglbasemixer.c | 0
 {ext => gst-libs/gst}/gl/gstglbasemixer.h | 3 +++
 {ext => gst-libs/gst}/gl/gstglmixer.c     | 2 ++
 {ext => gst-libs/gst}/gl/gstglmixer.h     | 4 +++-
 gst-libs/gst/gl/meson.build               | 4 ++++
 10 files changed, 21 insertions(+), 10 deletions(-)
 rename {ext => gst-libs/gst}/gl/gstglbasemixer.c (100%)
 rename {ext => gst-libs/gst}/gl/gstglbasemixer.h (99%)
 rename {ext => gst-libs/gst}/gl/gstglmixer.c (99%)
 rename {ext => gst-libs/gst}/gl/gstglmixer.h (99%)

diff --git a/ext/gl/gstglmosaic.c b/ext/gl/gstglmosaic.c
index 50578292..6b0ef638 100644
--- a/ext/gl/gstglmosaic.c
+++ b/ext/gl/gstglmosaic.c
@@ -42,6 +42,8 @@
 #include "config.h"
 #endif
 
+#include <gst/gl/gstglfuncs.h>
+
 #include "gstglelements.h"
 #include "gstglmosaic.h"
 #include "gstglutils.h"
diff --git a/ext/gl/gstglmosaic.h b/ext/gl/gstglmosaic.h
index 6bdb2492..731f523c 100644
--- a/ext/gl/gstglmosaic.h
+++ b/ext/gl/gstglmosaic.h
@@ -21,7 +21,7 @@
 #ifndef _GST_GL_MOSAIC_H_
 #define _GST_GL_MOSAIC_H_
 
-#include "gstglmixer.h"
+#include <gst/gl/gstglmixer.h>
 
 G_BEGIN_DECLS
 
diff --git a/ext/gl/gstglstereomix.h b/ext/gl/gstglstereomix.h
index 9971a891..530c3556 100644
--- a/ext/gl/gstglstereomix.h
+++ b/ext/gl/gstglstereomix.h
@@ -22,7 +22,7 @@
 #ifndef __GST_GL_STEREO_MIX_H__
 #define __GST_GL_STEREO_MIX_H__
 
-#include "gstglmixer.h"
+#include <gst/gl/gstglmixer.h>
 
 G_BEGIN_DECLS
 
@@ -64,7 +64,7 @@ struct _GstGLStereoMix
 {
   GstGLMixer mixer;
 
-  GLuint out_tex_id;
+  guint out_tex_id;
 
   GstGLViewConvert *viewconvert;
   GstGLStereoDownmix downmix_mode;
diff --git a/ext/gl/gstglvideomixer.h b/ext/gl/gstglvideomixer.h
index d07e1880..6c675cd1 100644
--- a/ext/gl/gstglvideomixer.h
+++ b/ext/gl/gstglvideomixer.h
@@ -21,7 +21,7 @@
 #ifndef _GST_GL_VIDEO_MIXER_H_
 #define _GST_GL_VIDEO_MIXER_H_
 
-#include "gstglmixer.h"
+#include <gst/gl/gstglmixer.h>
 
 G_BEGIN_DECLS
 
@@ -122,9 +122,9 @@ struct _GstGLVideoMixer
     GstGLShader *shader;
     GstGLShader *checker;
 
-    GLuint vao;
-    GLuint vbo_indices;
-    GLuint checker_vbo;
+    guint vao;
+    guint vbo_indices;
+    guint checker_vbo;
     GstGLMemory *out_tex;
 
     gboolean output_geo_change;
diff --git a/ext/gl/meson.build b/ext/gl/meson.build
index f1a94e2d..ad514014 100644
--- a/ext/gl/meson.build
+++ b/ext/gl/meson.build
@@ -30,14 +30,12 @@ opengl_sources = [
   'effects/gstgleffectblur.c',
   'effects/gstgleffectsobel.c',
   'effects/gstgleffectlaplacian.c',
-  'gstglbasemixer.c',
   'gstglcolorscale.c',
   'gstglcolorbalance.c',
   'gstgldeinterlace.c',
   'gstglfiltershader.c',
   'gstglfilterapp.c',
   'gstglmixerbin.c',
-  'gstglmixer.c',
   'gstglvideomixer.c',
   'gstglstereomix.c',
   'gstglstereosplit.c',
diff --git a/ext/gl/gstglbasemixer.c b/gst-libs/gst/gl/gstglbasemixer.c
similarity index 100%
rename from ext/gl/gstglbasemixer.c
rename to gst-libs/gst/gl/gstglbasemixer.c
diff --git a/ext/gl/gstglbasemixer.h b/gst-libs/gst/gl/gstglbasemixer.h
similarity index 99%
rename from ext/gl/gstglbasemixer.h
rename to gst-libs/gst/gl/gstglbasemixer.h
index c8234790..4440fc17 100644
--- a/ext/gl/gstglbasemixer.h
+++ b/gst-libs/gst/gl/gstglbasemixer.h
@@ -54,6 +54,7 @@ struct _GstGLBaseMixerPadClass
   GstVideoAggregatorPadClass parent_class;
 };
 
+GST_GL_API
 GType gst_gl_base_mixer_pad_get_type (void);
 
 #define GST_TYPE_GL_BASE_MIXER (gst_gl_base_mixer_get_type())
@@ -95,8 +96,10 @@ struct _GstGLBaseMixerClass
   gpointer _padding[GST_PADDING];
 };
 
+GST_GL_API
 GType gst_gl_base_mixer_get_type(void);
 
+GST_GL_API
 GstGLContext *      gst_gl_base_mixer_get_gl_context        (GstGLBaseMixer * mix);
 
 G_END_DECLS
diff --git a/ext/gl/gstglmixer.c b/gst-libs/gst/gl/gstglmixer.c
similarity index 99%
rename from ext/gl/gstglmixer.c
rename to gst-libs/gst/gl/gstglmixer.c
index 41c634d8..7d56662d 100644
--- a/ext/gl/gstglmixer.c
+++ b/gst-libs/gst/gl/gstglmixer.c
@@ -30,6 +30,8 @@
 
 #include "gstglmixer.h"
 
+#include <gst/gl/gstglfuncs.h>
+
 #define GST_CAT_DEFAULT gst_gl_mixer_debug
 GST_DEBUG_CATEGORY (gst_gl_mixer_debug);
 
diff --git a/ext/gl/gstglmixer.h b/gst-libs/gst/gl/gstglmixer.h
similarity index 99%
rename from ext/gl/gstglmixer.h
rename to gst-libs/gst/gl/gstglmixer.h
index 886e0094..773396d1 100644
--- a/ext/gl/gstglmixer.h
+++ b/gst-libs/gst/gl/gstglmixer.h
@@ -24,7 +24,6 @@
 #include <gst/gst.h>
 #include <gst/video/video.h>
 #include <gst/gl/gl.h>
-#include <gst/gl/gstglfuncs.h>
 #include "gstglbasemixer.h"
 
 G_BEGIN_DECLS
@@ -61,6 +60,7 @@ struct _GstGLMixerPadClass
   GstGLBaseMixerPadClass parent_class;
 };
 
+GST_GL_API
 GType gst_gl_mixer_pad_get_type (void);
 
 #define GST_TYPE_GL_MIXER (gst_gl_mixer_get_type())
@@ -102,8 +102,10 @@ struct _GstGLMixerClass
   GstGLMixerProcessTextures process_textures;
 };
 
+GST_GL_API
 GType gst_gl_mixer_get_type(void);
 
+GST_GL_API
 gboolean gst_gl_mixer_process_textures (GstGLMixer * mix, GstBuffer * outbuf);
 
 G_END_DECLS
diff --git a/gst-libs/gst/gl/meson.build b/gst-libs/gst/gl/meson.build
index 57b85b59..637898f6 100644
--- a/gst-libs/gst/gl/meson.build
+++ b/gst-libs/gst/gl/meson.build
@@ -15,6 +15,7 @@ gl_sources = files([
   'gstglapi.c',
   'gstglbasefilter.c',
   'gstglbasememory.c',
+  'gstglbasemixer.c',
   'gstglbasesrc.c',
   'gstglcolorconvert.c',
   'gstglbuffer.c',
@@ -30,6 +31,7 @@ gl_sources = files([
   'gstglframebuffer.c',
   'gstglmemory.c',
   'gstglmemorypbo.c',
+  'gstglmixer.c',
   'gstgloverlaycompositor.c',
   'gstglquery.c',
   'gstglrenderbuffer.c',
@@ -54,6 +56,7 @@ gir_gl_headers = files([
   'gstglapi.h',
   'gstglbasefilter.h',
   'gstglbasememory.h',
+  'gstglbasemixer.h',
   'gstglbasesrc.h',
   'gstglbuffer.h',
   'gstglbufferpool.h',
@@ -68,6 +71,7 @@ gir_gl_headers = files([
   'gstglframebuffer.h',
   'gstglmemory.h',
   'gstglmemorypbo.h',
+  'gstglmixer.h',
   'gstgloverlaycompositor.h',
   'gstglquery.h',
   'gstglrenderbuffer.h',
-- 
2.40.1

